tmt/package_managers/bootc.py

```diff
@@ -0,0 +1,262 @@
+        options: Optional[Options] = None,
+    ) -> ShellScript:
+        script = self.aux_engine.install_debuginfo(*installables, options=options)
+        self.containerfile_directives.append(f'RUN {script}')
+        return script
+
+
+@provides_package_manager('bootc')
+class Bootc(PackageManager[BootcEngine]):
+    NAME = 'bootc'
+
+    _engine_class = BootcEngine
+
+    # Invoke e.g. `bootc status`, and check if it returns "Booted image".
+    probe_command = ShellScript(
+        "type bootc && sudo bootc status && ((sudo bootc status --format yaml | grep -e 'booted: null' -e 'image: null') && exit 1 || exit 0)"  # noqa: E501
+    ).to_shell_command()
+
+    # Needs to be bigger than priorities of `yum`, `dnf`, `dnf5` and `rpm-ostree`.
+    probe_priority = 130
+
+    def check_presence(self, *installables: Installable) -> dict[Installable, bool]:
+        script = self.engine.check_presence(*installables)
+
+        if len(installables) == 1 and isinstance(installables[0], FileSystemPath):
+            try:
+                self.guest.execute(script)
+
+            except RunError as exc:
+                if exc.returncode == 1:
+                    return {installables[0]: False}
+
+                raise exc
+
+            return {installables[0]: True}
+
+        try:
+            output = self.guest.execute(script)
+            stdout = output.stdout
+
+        except RunError as exc:
+            stdout = exc.stdout
+
+        if stdout is None:
+            raise GeneralError("rpm presence check provided no output")
+
+        results: dict[Installable, bool] = {}
+
+        for line, installable in zip(stdout.strip().splitlines(), installables):
+            match = re.match(rf'package {re.escape(str(installable))} is not installed', line)
+            if match is not None:
+                results[installable] = False
+                continue
+
+            match = re.match(rf'no package provides {re.escape(str(installable))}', line)
+            if match is not None:
+                results[installable] = False
+                continue
+
+            results[installable] = True
+
+        return results
+
+    def build_container(self) -> None:
+        image_tag = f"localhost/tmt-deps-{uuid.uuid4()}"
+
+        # Write the final Containerfile
+        containerfile_dir = Path(tempfile.gettempdir(), str(uuid.uuid4()))
+        containerfile_path = Path(containerfile_dir, 'Containerfile')
+
+        containerfile = '\n'.join(self.engine.containerfile_directives)
+
+        base_image = self.engine._get_current_bootc_image()
+
+        try:
+            # Check if image exists and store it from bootc of don't
+            self.guest.execute(
+                ShellScript(
+                    f'podman images {base_image} | grep {base_image} || bootc image copy-to-storage --target {base_image}'  # noqa: E501
+                )
+            )
+            self.guest.execute(ShellScript(f'mkdir -p {containerfile_dir!s}'))
```

Comment:
You will need different paths for runners and guests. Or put the remote path into the current workdir. The path generated by `tempfile.gettempdir()` is node-specific; it was unused and available on the runner, but it may very easily be used, by a sheer accident, by something completely unrelated on the guest. So either use path that is truly same on both side, like workdir, or you need different temporary directory on thee guest, e.g. `Guest.mkdtemp`.

Reply:
Right, I was waiting for the `Guest.mkdtemp` implementation. I used it to create a temporary directory